clients/venice-admin-tool/src/main/java/com/linkedin/venice/datarecovery/StoreRepushCommand.java

```diff
@@ -0,0 +1,366 @@
+package com.linkedin.venice.datarecovery;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.apache.commons.lang.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+
+/**
+ * StoreRepushCommand contains the details of executing/processing repush command.
```

Comment:
Can you add the contract with the underlying tool as javadoc?

Reply:
I recommend you look at `repush` and `venice-tools` with different lenses. We are adding `repush` as a Venice concept and `venice-tools` is internal tooling that helps achieve that.

Think of `repush` from a OSS Venice user's perspective and see what options would you expect it to offer, and the arguments that would be ideal for it. LinkedIn tooling should not be considered first class users of Venice in OSS (just another adopter of it). We need to build abstractions that can work for the general system and build implementations of those abstractions internally. Tying OSS Venice arguments to `venice-tools` arguments tightly couples these systems and doesn't offer enough flexibility.

To me, looking from this lens, there are quite a few things should not be present in OSS Venice codebase.

Here are some:
1. `REPUSH_MODULE`
2. `REPUSH_SOURCE_KAFKA`
3. `Password+VIP`
4. All the `Params` that are not needed by the data-recovery tooling directly. (You can add a new argument `EXTRA_COMMAND_ARGS` that forwards any args specified to the underlying tooling.)
5. `matchAzkabanSuccessPattern` and `matchAzkabanFailurePattern`

Instead of calling `venice-tools` directly, can we proxy it via a different command (like an internal python/shell script) that can hide all the internal aspects of the tools and offer a cleaner interface that works better for OSS code?

This internal script would adhere to the contract defined by OSS Venice and translates the instructions to those appropriate for `venice-tools`. All `expect` logic can be handled by that proxy tool and that would prevent us from leaking any peculiarities of internal tools into open-source

Reply:
> Here are some:
> 
> 1. REPUSH_MODULE
> 2. REPUSH_SOURCE_KAFKA
> 3. Password+VIP
> 4. All the Params that are not needed by the data-recovery tooling directly. (You can add a new argument EXTRA_COMMAND_ARGS that forwards any args specified to the underlying tooling.)
> 5. matchAzkabanSuccessPattern and matchAzkabanFailurePattern

1, 2, 4, makes sense to me and I can fix them in this change. For point 5, as I explained in the above paragraph, we probably want to wait the formalization work done at the `venice-tool` side, so that we can get rid of it. As for point 3, maybe we can simplify the `venice-tool` side directly, by making it to take credentials as a parameter instead of always showing a prompt for typing from user. Let me know what you are thinking and if that makes sense, can I postpone to address it in the next change?

Reply:
> Unfortunately, today it is not organized that way, so that's why we need matchAzkabanSuccessPattern and matchAzkabanFailurePattern to parse the output. It would need changes on the venice-tool side, and once it is done, we can define the contract that is more concisely.

This is exactly what I mean by decoupling `venice-tools` from OSS Venice. If `venice-tools repush` is expected to be a first-class implementation of data recovery repush, then we need to follow the order:
1. Define the contract for repush from the perspective of OSS Venice and only code for this contract
2. Make changes to `venice-tools repush` to adhere to this contract
    * We should not add code in OSS Venice that is specific to `venice-tools` in it's current state as it does not adhere to the contract.

Regarding `Password+VIP`: Taking credentials as a parameter is possible but it is a security risk (as it will show up in bash history), so, we will need to continue supporting both ways (as an arg and a prompt)

Reply:
Thanks for the offline discussion Nisarg. As we discussed, I will

1. modify this PR to define a contract between OSS venice and `venice-tools`. Thus, this change is coding against this contract, though some of it is not fully supported by `venice-tools` yet.
2. as the next step, make changes to `venice-tools repush` so that it comply to this contract as we..

We might get away with the security issue that you worried, as `history` might only display the typed command (https://tldp.org/LDP/GNU-Linux-Tools-Summary/html/x1712.htm). However, maybe the best solution is to execute using a headless account where it doesn't require inputing a user credential at all. Will follow up on this point anyway.